packages/app-config-writer/src/preview-config/ui5-yaml.ts

```diff
@@ -0,0 +1,306 @@
+}
+
+/**
+ * Updates the preview middleware configurations according to the scripts they are being used in package.json.
+ *
+ * It will process all given UI5 configuration yaml files and check if the preview middleware configuration must be updated based on a given script.
+ * If a script is valid, the preview middleware configuration will be updated and the corresponding file renamed.
+ * UI5 configuration yaml files which are not used in any script will be sanitized.
+ *
+ * @param fs - file system reference
+ * @param basePath - base path to be used for the conversion
+ * @param logger logger to report info to the user
+ */
+export async function updatePreviewMiddlewareConfigs(
+    fs: Editor,
+    basePath: string,
+    logger?: ToolsLogger
+): Promise<void> {
+    const ui5YamlFileNames = await getAllUi5YamlFileNames(basePath, fs);
+    const unprocessedUi5YamlFileNames = [...ui5YamlFileNames];
+    const packageJsonPath = join(basePath, 'package.json');
+    const packageJson = fs.readJSON(packageJsonPath) as Package | undefined;
+    for (const [scriptName, script] of Object.entries(packageJson?.scripts ?? {})) {
+        if (!script || !isValidPreviewScript(scriptName, script)) {
+            continue;
+        }
+
+        const ui5Yaml = basename(extractYamlConfigFileName(script));
+        unprocessedUi5YamlFileNames.splice(unprocessedUi5YamlFileNames.indexOf(ui5Yaml), 1);
+
+        if (
+            !isUi5YamlToBeConverted(ui5Yaml, scriptName, ui5YamlFileNames, logger) ||
+            (await isUi5YamlAlreadyConverted(fs, basePath, ui5Yaml, scriptName, script, logger))
+        ) {
+            continue;
+        }
+
+        try {
+            await processUi5YamlConfig(fs, basePath, ui5Yaml, script);
+        } catch (error) {
+            logger?.warn(
+                `Skipping script, '${scriptName}', which refers to the UI5 YAML configuration file '${ui5Yaml}'. ${error.message}`
+            );
+            continue;
+        }
+
+        const { path } = extractUrlDetails(script);
+        if (path) {
+            await renameSandbox(fs, join(await getWebappPath(basePath), path), logger);
+        }
+        ensurePreviewMiddlewareDependency(packageJson, fs, packageJsonPath);
+
+        logger?.info(
+            `The UI5 YAML configuration file, '${ui5Yaml}', has been updated according to script, '${scriptName}'.`
+        );
+    }
+    for (const ui5Yaml of unprocessedUi5YamlFileNames) {
+        //at least adjust deprecated preview config of unused ui5 yaml configurations
+        try {
+            await processUi5YamlConfig(fs, basePath, ui5Yaml, '', true);
+        } catch (error) {
+            logger?.warn(`Skipping UI5 yaml configuration file '${ui5Yaml}'. ${error.mesage}`);
+        }
+        logger?.warn(
+            `The UI5 YAML configuration file, '${ui5Yaml}', is not used in any preview script. Outdated preview middleware will be adjusted if necessary.`
```

Comment:
```suggestion
            `The UI5 YAML configuration file '${ui5Yaml}', is not used in any preview script. Outdated preview middleware will be adjusted, if necessary.`
```

Reply:
Done in f41acea6cdc7bcf341fd9c72b7a4306e0012a3b5